### PR TITLE
First example of dependency injection

### PR DESCRIPTION
**Pull request by nstapelbroek**

### CODE REVIEW

File: src/AppBundle/Service/D6Dice.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+class D6Dice implements Dice {
+
+    /**
+     * D6Dice constructor. Fills the dice with the 6 values
+     * @param array $values
+     * @throws IllegalArgumentException
+     */
+    public function __construct(array $values)
+    {
+        if(count($values) != 61) {
```
Comment (adri) [inline src/AppBundle/Service/D6Dice.php]: typo?

File: src/AppBundle/Controller/DiceController.php [hunk 2]
```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace AppBundle\Controller;
+
+use AppBundle\Contract\Dice;
```
Comment (adri) [inline src/AppBundle/Controller/DiceController.php]: u should suffix the interface name with interface. e.g. DiceInterface

File: src/AppBundle/Resources/config/services.xml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" ?>
+<container xmlns="http://symfony.com/schema/dic/services"
+           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+           xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+    <services>
+        <service id="d6dice_service" class="AppBundle\Service\D6Dice" />
```
Comment (adri) [inline src/AppBundle/Resources/config/services.xml]: don't forge to set the arguments of the dice.

Comment (adri) [inline src/AppBundle/Resources/config/services.xml]: rename it to a convention: "app.dice.d6"

File: src/AppBundle/Service/D6Dice.php (hunk 1 above)
Comment (adri) [inline src/AppBundle/Service/D6Dice.php]: it should already error here, please clear you cache using `php ./bin/console cache:clear`

File: web/app_dev.php [hunk 4]
```diff
@@ -10,13 +10,6 @@
 
 // This check prevents access to debug front controllers that are deployed by accident to production servers.
 // Feel free to remove this, extend it, or make something more sophisticated.
-if (isset($_SERVER['HTTP_CLIENT_IP'])
```
Comment (adri) [inline web/app_dev.php]: ok, you don't need your adjustments to the .htaccess etc. just run your app using `php ./bin/console server:run`

File: composer.json [hunk 5]
```diff
@@ -26,7 +26,8 @@
         "symfony/monolog-bundle": "^2.8",
         "sensio/distribution-bundle": "^5.0",
         "sensio/framework-extra-bundle": "^3.0.2",
-        "incenteev/composer-parameter-handler": "^2.0"
+        "incenteev/composer-parameter-handler": "^2.0",
+        "symfony/dependency-injection": "^3.0"
```
Comment (adri) [inline composer.json]: not necessary

**Comment by nstapelbroek** [comment]
Could you change all your array notations to the short notation? Thanks

**Comment by renatomefidf** [comment]
@nstapelbroek lol

**Comment by renatomefidf** [comment]
Common, sudo merge? seriously?

**Comment by nstapelbroek** [comment]
Fixed it :see_no_evil:

**Comment by nstapelbroek** [comment]
Any plans of merging the PR's anyway? Or would you like some improvements

**Comment by renatomefidf** [comment]
Go ahead :p

**Pull request merged**